tests/run.go

```diff
@@ -398,6 +398,136 @@ func Run(o *option.Option) {
+				// /blob/ce07bb28eadc18183df15ca5346293af6b020b33/integration/tests/api/docker_test.go#L216-L222
+				// Ref. https://github.com/google/cadvisor/blob/master/integration/tests/api/docker_test.go#L216-L222
+				cpuWeight := command.StdoutStr(o, "run", "--rm", "--cpu-shares", "2000",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpu.weight")
+				gomega.Expect(cpuWeight).To(gomega.Equal("77"))
+			})
+
+			// assume the host has at least 2 CPUs
+			ginkgo.When("--cpuset-cpus is used", func() {
+				ginkgo.It("should set CPUs in which to allow execution as cpu 1 with --cpuset-cpus=1", func() {
+					cpuSet := command.StdoutStr(o, "run", "--cpuset-cpus", "1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("1"))
+				})
+
+				ginkgo.It("should set CPUs in which to allow execution as cpu 0-1 with --cpuset-cpus=0-1", func() {
+					cpuSet := command.StdoutStr(o, "run", "--cpuset-cpus", "0-1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("0-1"))
+				})
+
+				ginkgo.It("should set CPUs in which to allow execution as cpu 0-1 with --cpuset-cpus=0,1", func() {
+					cpuSet := command.StdoutStr(o, "run", "--cpuset-cpus", "0,1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("0-1"))
+				})
+			})
+
+			// The range form (e.g., `--cpuset-mems 0-1`) is not tested because maybe there's only one memory node (i.e., `0`) on the host,
+			// and if that's the case, any number other than 0 would incur an error:
+			// https://man7.org/linux/man-pages/man7/cpuset.7.html#WARNINGS
+			ginkgo.It("should set memory nodes (MEMs) in which to allow execution as memory node 0 with --cpuset-mems=0", func() {
+				cpuMems := command.StdoutStr(o, "run", "--cpuset-mems", "0",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.mems")
+				gomega.Expect(cpuMems).To(gomega.Equal("0"))
+			})
+
+			ginkgo.It("should set the memory limit with --memory", func() {
+				mem := command.StdoutStr(o, "run", "--memory", "42m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.max")
+				gomega.Expect(mem).To(gomega.Equal("44040192"))
+			})
+
+			ginkgo.It("should set the memory soft limit with --memory-reservation", func() {
+				mem := command.StdoutStr(o, "run", "--memory-reservation", "6m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.low")
+				gomega.Expect(mem).To(gomega.Equal("6291456"))
+			})
+
+			ginkgo.It("should set the amount of memory this container is allowed to swap to disk with --memory-swap", func() {
+				mem := command.StdoutStr(o, "run", "--memory", "42m", "--memory-swap", "100m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.max", "memory.swap.max")
+				gomega.Expect(mem).To(gomega.Equal("44040192\n60817408"))
+			})
+
+			// TODO: --memory-swappiness --oom-kill-disable --oom-score-adj --blkio-weight --cgroupns
```

Comment:
- `--oom-score-adj`: Could we test 100 (min) and 1000 (max)? https://github.com/containerd/nerdctl/pull/1521/files
- ` --blkio-weight`: Maybe we can test it now? https://github.com/containerd/nerdctl/issues/1514
- `--memory-swappiness`: Could we explain that currently we don't plan to support cgroupv1-only flags so that we're not testing it?
- `--oom-kill-disable`: Could we link https://github.com/containerd/nerdctl/issues/1520 so that folks know that we are waiting for it to be implemented for cgroupv2, and then we can test it?

We can probably also move these TODOs to the end of the cgroup block (i.e., where the explanation of `--device` lives now) to make it clearer.